### ISSUE TITLE
I want to replace SIFT  extraction in COLMAP

### ISSUE THREAD

**Issue by wangjinhoon** [colmap/colmap#2652]
Hello, thank you for providing me with a good code. 

I want to replace SIFT (Scale-Invariant Feature Transform) extraction in COLMAP with another method. Is it possible to extract keypoints and descriptors using Python, save them, and then input them into COLMAP for 3D reconstruction?

or Can you tell me about the related tutorial?

Apologies if I used an incorrect method in creating the issue

**Environment:**
 - OS: linux 20.04
 - COLMAP Version : latest version

**Comment by yuancaimaiyi** [comment]
@wangjinhoon  please use https://github.com/cvg/Hierarchical-Localization ,you can easily replace sift!

**Comment by biggiantpigeon** [comment]
yes you can, this repo totally do this to evaluate different features. Its colmap version may be old but the method is the same:
https://github.com/ubc-vision/image-matching-benchmark

**Comment by wangjinhoon** [comment]
@yuancaimaiyi Thank you for your response. This is what I wanted. Let me try this

**Comment by ahojnnes** [comment]
Closing: replacing SIFT answered; hloc and image-matching-benchmark suggested.
